File: ws/src/main/scala/de/thm/ii/fbs/services/checker/CheckerService.scala
package de.thm.ii.fbs.services.checker

import de.thm.ii.fbs.model.{CheckrunnerConfiguration, User}

/**
  * The trait representing the checker
  */
trait CheckerService {
  /**
    * Notify about the new submission
    * @param taskID the taskID for the submission
    * @param submissionID the id of the sumission
    * @param cc the check runner of the sumission
    * @param fu the user which triggered the sumission
    */
  def notify(taskID: Int, submissionID: Int, cc: CheckrunnerConfiguration, fu: User): Unit
}

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/services/v2/checker/trait/CheckerService.kt
package de.thm.ii.fbs.services.v2.checker.trait

import de.thm.ii.fbs.model.v2.CheckrunnerConfiguration
import de.thm.ii.fbs.model.v2.User

open class CheckerService {
    /**
     * Notify about the new submission
     *
     * @param taskID       the taskID for the submission
     * @param submissionID the id of the sumission
     * @param cc           the check runner of the sumission
     * @param fu           the user which triggered the sumission
     */
    open fun notify(taskID: Int, submissionID: Int, cc: CheckrunnerConfiguration, fu: User) {}
}